lib/iris/util.py

```diff
@@ -1517,6 +1517,111 @@ def demote_dim_coord_to_aux_coord(cube, name_or_coord):
     cube.add_aux_coord(dim_coord, coord_dim)
 
 
+def flatten_multidim_coord(cube, name_or_coord):
+    """
+    Flatten the Aux coord and associated dimensions on the cube
+
+
+    * cube
+        An instance of :class:`iris.cube.Cube`
+
+    * name_or_coord:
+        Either
+
+        (a) An instance of :class:`iris.coords.DimCoord`
+
+        or
+
+        (b) the :attr:`standard_name`, :attr:`long_name`, or
+        :attr:`var_name` of an instance of an instance of
+        :class:`iris.coords.DimCoord`.
+
+    For example::
+
+        >>> print(cube)
+        rainfall_rate / (kg m-2 s-1)        (x: 5; y: 3)
+             Dimension coordinates:
+                  x                           x     -
+                  y                           -     x
+             Auxiliary coordinates:
+                  latitude                    x     x
+                  longitude                   x     x
+
+        >>> flatten_multidim_coord(cube, 'longitude')
+        >>> print(cube)
+        rainfall_rate / (kg m-2 s-1)        (--: 15)
+             Auxiliary coordinates:
+                  latitude                    x
+                  longitude                   x
+    """
+    import numpy as np
+    from iris.cube import Cube
+
+    if isinstance(name_or_coord, six.string_types):
+        coord = cube.coord(name_or_coord)
+    elif isinstance(name_or_coord, iris.coords.Coord):
+        coord = name_or_coord
+    else:
+        # Don't know how to handle this type
+        msg = ("Don't know how to handle coordinate of type {}. "
+               "Ensure all coordinates are of type six.string_types or "
+               "iris.coords.Coord.")
+        msg = msg.format(type(name_or_coord))
+        raise TypeError(msg)
+
+    coord_dims = cube.coord_dims(coord)
+
+    if len(coord_dims) == 1:
+        # Do nothing
+        return cube
+
+    other_dims = list(set(range(cube.ndim)) - set(coord_dims))
+    shape_array = np.asarray(cube.shape)
+    new_shape = np.hstack([shape_array[other_dims],
+                           np.product(shape_array[list(coord_dims)])])
+
+    # Get all the coords which span the dimensions to be flattened
+    coords_to_flatten = cube.coords(dimensions=coord_dims)
+    # And any other coords which span any one of the dimensions
+    # These (1-D) coordinates are expanded before being flattened
+    coords_to_expand = []
+    for d in coord_dims:
+        coords_to_expand.extend(cube.coords(dimensions=d))
+
+    # Expand the coords which need expanding first, then add to flatten list
+    for c in coords_to_expand:
+        new_points = broadcast_to_shape(c.points, coord.shape, (0,))
```

Comment:
The `(0,)` here will need generalising I think.

Reply:
Actually, although it looks a bit wierd the zero here is fine because c is must be one dimensional. I've added a comment to clarify

Reply:
`c` is one dimensional, but `coord` is multidimensional, and the tuple indicates the dimension of `coord` that `c` is getting mapped to.  So, for your docstring example, you would have 
`broadcast_to_shape(c.points, (2, 2), (0,))` for the `x` coord, but 
`broadcast_to_shape(c.points, (2, 2), (1,))` for the `y` coord.

Reply:
Ah, yes - sorry! :flushed:

Is there a nice way of getting the dimension of an Aux Coord for a given DimCoord?

Reply:
No specific method that I know of.  I'd be tempted to combine the loop with the one at line 1614.

```python
for dim_map, d in enumerate(coord_dims):
    for c in cube.coords(dimensions=d):
        new_points = broadcast_to_shape(c.points, coord.shape, (dim_map,))
        ...etc...
```

Reply:
Good idea!